### PR TITLE
@types/lunr: Updated Builder functions to add new optional arguments

### PR DESCRIPTION
**Pull request by HokieGeek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/olivernn/lunr.js/blob/master/lunr.js#L2423
https://github.com/olivernn/lunr.js/blob/master/lunr.js#L2477
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/lunr/index.d.ts [hunk 1]
```diff
@@ -127,9 +127,14 @@ declare namespace lunr {
...
          *
+         * Fields can be boosted at build time. This allows terms within that field to have more
+         * importance when ranking search results. Use a field boost to specify that matches
+         * within one field are more important than other fields.
+         *
          * @param field - The name of a field to index in all documents.
+         * @param attributes - Optional attributes associated with this field.
          */
-        field(field: string): void;
+        field(field: string, attributes?: object): void;
```
Comment (HokieGeek) [inline types/lunr/index.d.ts]: Used `object` due to attributes being set as a value of `_fields` which is just an object.

https://github.com/olivernn/lunr.js/blob/master/lunr.js#L2423
```javascript
this._fields[fieldName] = attributes || {}
```
https://github.com/olivernn/lunr.js/blob/master/lunr.js#L2360
```javascript
lunr.Builder = function () {
  this._ref = "id"
  this._fields = Object.create(null)
  this._documents = Object.create(null)
```

File: types/lunr/index.d.ts [hunk 2]
```diff
@@ -160,9 +165,13 @@ declare namespace lunr {
...
          * cause errors.
          *
+         * Entire documents can be boosted at build time. Applying a boost to a document indicates that
+         * this document should rank higher in search results than other documents.
+         *
          * @param doc - The document to add to the index.
+         * @param attributes - Optional attributes associated with this document.
          */
-        add(doc: object): void;
+        add(doc: object, attributes?: object): void;
```
Comment (HokieGeek) [inline types/lunr/index.d.ts]: Used `object` due to attributes being set as a value of `_documents` which is just an object.

https://github.com/olivernn/lunr.js/blob/master/lunr.js#L2477
```javascript
this._documents[docRef] = attributes || {}
```
https://github.com/olivernn/lunr.js/blob/master/lunr.js#L2360
```javascript
lunr.Builder = function () {
  this._ref = "id"
  this._fields = Object.create(null)
  this._documents = Object.create(null)
```

File: types/lunr/index.d.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-// Type definitions for lunr.js 2.1
+// Type definitions for lunr.js 2.3
```
Comment (HokieGeek) [inline types/lunr/index.d.ts]: The modified functions were introduced in 2.3.0 with this commit:

https://github.com/olivernn/lunr.js/commit/95d585298524cce52ab29c96f74ee8961d2ea9b8#diff-0fb61214341893489a1490be5f5b65a7

**Review by amcasey — APPROVED** [review]
Thanks for the links to the underlying project!

**Comment by HokieGeek** [comment]
> Thanks for the links to the underlying project!
Of course! Silly not to. Thanks for taking a look at it!

**Pull request merged**